src/num/theories/cv_compute/automation/cv_primScript.sml

```diff
@@ -904,10 +904,15 @@ QED
 Theorem cv_rep_word_lsr[cv_rep]:
   from_word (word_lsr (w:'a word) n)
   =
-  cv_div (from_word w) (cv_exp (Num 2) (Num n))
+  cv_if (cv_lt (Num n) (Num (dimindex (:'a))))
```

Comment:
The expression corresponding to Num n is duplicated on the right-hand side of this equation meaning it'll be computed twice. You'll need to let bind Num n on the right-hand side to avoid this duplication, ie 

... = let k = Num n in ... k ...

Reply:
Done